### ISSUE TITLE
Parametertree widgets may not respect limits of associated Parameter

### ISSUE THREAD

**Issue by polwel** [pyqtgraph/pyqtgraph#486]
A bug in `pyqtgraph.parametertree.parameterTypes.WidgetParameterItem` will make the spinboxes in the parametertree randomly disregard the bounds of the associated parameter. 

Currently, the `makeWidget()` method looks like this:
```python
defs = {
    'value': 0, 'min': None, 'max': None,
    'step': 1.0, 'dec': False,
    'siPrefix': False, 'suffix': '', 'decimals': 3,
}
if 'limits' in opts:
    defs['bounds'] = opts['limits']
w = SpinBox()
w.setOpts(**defs)
```

If `limits` is specified in the `opts`, the spinbox will be passed both a `bounds` option, as well as the invalid `min` and `max` options. Those should not be passed if `limits` are given.
`Spinbox` will loop over the options, and depending on which one it processes first (which is random, since a `dict` is not ordered), it may use the wrong one.

**Solution**: Replace the above by 
``` python
...
if 'limits' in opts:
    defs['bounds'] = opts['limits']
    del defs['min']
    del defs['max']
...
```

**Comment by polwel** [comment]
I now see this has already been fixed in the develop branch. Closing.

**Comment by polwel** [comment]
FYI, the fix was committed in 670d63cdf443d667eece3b203083692588f41693.
